variables.tf

```diff
@@ -94,6 +100,90 @@ variable "managed_identities" {
   default     = {}
   description = "Managed identities to be created for the resource."
 }
+variable "key_vault_secrets_provider_enabled" {
+  type        = bool
+  default     = false
+  description = "(Optional) Whether to use the Azure Key Vault Provider for Secrets Store CSI Driver in an AKS cluster. For more details: https://docs.microsoft.com/en-us/azure/aks/csi-secrets-store-driver"
+  nullable    = false
+}
+
+variable "log_analytics_workspace_enabled" {
+  type        = bool
+  default     = false
+  description = "Enable the integration of azurerm_log_analytics_workspace and azurerm_log_analytics_solution: https://docs.microsoft.com/en-us/azure/azure-monitor/containers/container-insights-onboard"
+  nullable    = false
+}
+
+variable "monitor_metrics" {
+  type = object({
+    annotations_allowed = optional(string)
+    labels_allowed      = optional(string)
+  })
+  default     = null
+  description = <<-EOT
+  (Optional) Specifies a Prometheus add-on profile for the Kubernetes Cluster
+  object({
+    annotations_allowed = "(Optional) Specifies a comma-separated list of Kubernetes annotation keys that will be used in the resource's labels metric."
+    labels_allowed      = "(Optional) Specifies a Comma-separated list of additional Kubernetes label keys that will be used in the resource's labels metric."
+  })
+EOT
+}
+
+variable "msi_auth_for_monitoring_enabled" {
+  type        = bool
+  default     = null
+  description = "(Optional) Is managed identity authentication for monitoring enabled?"
+}
+
+variable "node_pools" {
+  type = map(object({
+    vm_size             = string
+    enable_auto_scaling = bool
+    max_count           = number
+    min_count           = number
+    node_count          = number
+    os_sku              = string
+    mode                = optional(string, "User")
+    os_disk_size_gb     = optional(number, null)
+    tags                = optional(map(string), {})
+    zones               = optional(string)
+  }))
+  default = {
+    "1" = {
+      vm_size             = "Standard_D4d_v5"
+      enable_auto_scaling = true
+      max_count           = 110
+      min_count           = 2
+      node_count          = 2
+      os_sku              = "Ubuntu"
+      mode                = "User"
+    },
+    "2" = {
+      vm_size             = "Standard_D4d_v5"
+      enable_auto_scaling = true
+      max_count           = 110
+      min_count           = 2
+      node_count          = 2
+      os_sku              = "Ubuntu"
+      mode                = "User"
+    },
+    "3" = {
+      vm_size             = "Standard_D4d_v5"
+      enable_auto_scaling = true
+      max_count           = 110
+      min_count           = 2
+      node_count          = 2
+      os_sku              = "Ubuntu"
+      mode                = "User"
+    }
+  }
+  description = "The node pools to create on the Kubernetes Cluster."
+
+  validation {
+    condition     = length(keys(var.node_pools)) >= 3
```

Comment:
1. Should we add a `null` check here? Would `length` function throw an error when `var.node_pools` is `null`?
2. Could we simplify the length check expression to `length(var.node_pools) >= 3`?

Reply:
Thanks @lonegunmanb  for reviewing
 Thanks for catching this both are great ideas and i've added your suggestions